src/mondefense.cpp

```diff
@@ -47,41 +47,75 @@ void mdefense::zapback( monster *const m, Creature *const source, projectile con
     source->check_dead_state();
 }
 
-static int posp( int posc ) //Determines orientation of target from the source
+static int sign( int arg )
 {
-    if( posc > 0 ) {
+    if( arg > 0 ) {
         return 1;
     }
-    if( posc < 0 ) {
+
+    if( arg < 0 ) {
         return -1;
-    } else {
-        return 0;
     }
+
+    return 0;
 }
 
 void mdefense::acidsplash( monster *const m, Creature *const source, projectile const *const proj )
 {
-    if( source == nullptr || m == nullptr ) {
+    // Would be useful to have the attack data here, for cutting vs. bashing etc.
+    if( m == nullptr ) {
         return;
     }
-    if( proj != nullptr && rng( 0, 40 ) > m->def_chance ) {
+
+    if( proj != nullptr && !one_in( 3 ) ) {
         return; //Less likely for a projectile to deliver enough force
     }
 
-    int dx = posp( source->posx() - m->posx() );
-    int dy = posp( source->posy() - m->posy() );
-    if( one_in( 2 ) ) {
-        for( int i = 0; i < rng( 3, 5 ); i++ ) {
-            const int mul = one_in( 2 ) ? 2 : 1;
-            tripoint dest( m->posx() + ( dx * mul ) + rng( -1, 1 ),
-                           m->posy() + ( dy * mul ) + rng( -1, 1 ),
-                           m->posz() );
-            g->m.add_field( dest, fd_acid, rng( 2, 3 ), 0 );
+    size_t num_drops = rng( 2, 4 );
+    player const *const foe = dynamic_cast<player *>( source );
+    if( proj == nullptr && foe != nullptr ) {
+        if( foe->weapon.is_cutting_weapon() ) {
+            num_drops += rng( 1, 2 );
+        }
+
+        if( foe->unarmed_attack() ) {
+            damage_instance const burn {
+                DT_ACID, static_cast<float>( rng( 1, 5 ) )
+            };
+
+            if( one_in( 2 ) ) {
```

Comment:
Future note, unarmed attacks should pick the attacking limb so that styles can adjust this.